packages/next/src/lib/worker.ts

```diff
@@ -24,11 +24,36 @@ export class Worker {
     this._worker = undefined
 
     const createWorker = () => {
-      this._worker = new JestWorker(workerPath, farmOptions) as JestWorker
+      this._worker = new JestWorker(workerPath, {
+        ...farmOptions,
+        forkOptions: {
+          ...farmOptions.forkOptions,
+          env: {
+            ...((farmOptions.forkOptions?.env || {}) as any),
+            ...process.env,
+            // we don't pass down NODE_OPTIONS as it can
+            // extra memory usage
+            NODE_OPTIONS: '',
+          } as any,
+        },
+      }) as JestWorker
       restartPromise = new Promise(
         (resolve) => (resolveRestartPromise = resolve)
       )
 
+      for (const worker of ((this._worker as any)._workerPool?._workers ||
+        []) as {
+        _child: ChildProcess
+      }[]) {
+        worker._child.on('exit', (code, signal) => {
```

Comment:
do you have docs handy for this lib/event? I'm going to look now, just curious for my own edification.

Reply:
https://nodejs.org/api/child_process.html